presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java

```diff
@@ -744,27 +748,27 @@ private AmazonS3 createAmazonS3Client(Configuration hadoopConfig, ClientConfigur
         }
     }
 
-    private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configuration conf)
+    private Optional<AWSCredentialsProvider> createAwsCredentialsProvider(URI uri, Configuration conf)
```

Comment:
What about access to Glue Metastore?

I think `io.prestosql.plugin.hive.s3.PrestoS3FileSystem#createAwsCredentialsProvider` should be extracted, properly parametrized and also used in `io.prestosql.plugin.hive.metastore.glue.GlueHiveMetastore#createAsyncGlueClient`

Reply:
I will fix it too. I would rather avoid factoring them in because a customer should be able to use different AWS accounts for each of these.

Reply:
> customer should be able to use different AWS accounts for each of these.

That should be the case anyway. What I meant is that IAM feature toggles should be similar for both S3 and Glue